### PR TITLE
Use the unique user id for user session APIs

### PR DESCRIPTION
**Pull request by pamodaaw**
Fix wso2/product-is#7725

### CODE REVIEW

File: components/org.wso2.carbon.identity.api.user.common/src/main/java/org/wso2/carbon/identity/api/user/common/Util.java [hunk 1]
```diff
@@ -82,4 +90,54 @@ public static String resolveUserIdFromUser(User user) {
...
+
+        try {
+            isUserValid = UserSessionStore.getInstance().isExistingUser(userId);
+            if (!isUserValid) {
+                UniqueIDUserStoreManager uniqueIdEnabledUserStoreManager =
+                        getUniqueIdEnabledUserStoreManager(realmService, tenantDomain);
+                isUserValid = uniqueIdEnabledUserStoreManager.isExistingUserWithID(userId);
+            }
+        } catch (org.wso2.carbon.user.api.UserStoreException | UserSessionException e) {
+
```
Comment (nilasini) [inline components/org.wso2.carbon.identity.api.user.common/src/main/java/org/wso2/carbon/identity/api/user/common/Util.java]: remove extra new line

File: components/org.wso2.carbon.identity.api.user.common/src/main/java/org/wso2/carbon/identity/api/user/common/Util.java [hunk 2]
```diff
@@ -82,4 +90,54 @@ public static String resolveUserIdFromUser(User user) {
...
+        boolean isUserValid;
+
+        try {
+            isUserValid = UserSessionStore.getInstance().isExistingUser(userId);
+            if (!isUserValid) {
+                UniqueIDUserStoreManager uniqueIdEnabledUserStoreManager =
+                        getUniqueIdEnabledUserStoreManager(realmService, tenantDomain);
+                isUserValid = uniqueIdEnabledUserStoreManager.isExistingUserWithID(userId);
+            }
+        } catch (org.wso2.carbon.user.api.UserStoreException | UserSessionException e) {
```
Comment (nilasini) [inline components/org.wso2.carbon.identity.api.user.common/src/main/java/org/wso2/carbon/identity/api/user/common/Util.java]: Can't u straightly call UserStoreException without the package names?

File: components/org.wso2.carbon.identity.api.user.session/org.wso2.carbon.identity.api.user.session.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/session/v1/core/SessionManagementService.java [hunk 3]
```diff
@@ -70,12 +70,51 @@
...
+     *
+     * @param userId unique id of the user
+     * @param limit  limit (optional)
+     * @param offset offset (optional)
+     * @param filter filter (optional)
+     * @param sort   sort (optional)
+     * @return SessionsDTO
+     */
+    public SessionsDTO getSessionsByUserId(String userId, Integer limit, Integer offset, String filter, String
+            sort) {
```
Comment (nilasini) [inline components/org.wso2.carbon.identity.api.user.session/org.wso2.carbon.identity.api.user.session.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/session/v1/core/SessionManagementService.java]: shall we fomat this?

File: components/org.wso2.carbon.identity.api.user.common/src/main/java/org/wso2/carbon/identity/api/user/common/Util.java (hunk 1 above)
Comment (pamodaaw) [reply components/org.wso2.carbon.identity.api.user.common/src/main/java/org/wso2/carbon/identity/api/user/common/Util.java]: fixed

File: components/org.wso2.carbon.identity.api.user.common/src/main/java/org/wso2/carbon/identity/api/user/common/Util.java (hunk 2 above)
Comment (pamodaaw) [reply components/org.wso2.carbon.identity.api.user.common/src/main/java/org/wso2/carbon/identity/api/user/common/Util.java]: updated

File: components/org.wso2.carbon.identity.api.user.session/org.wso2.carbon.identity.api.user.session.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/session/v1/core/SessionManagementService.java (hunk 3 above)
Comment (pamodaaw) [reply components/org.wso2.carbon.identity.api.user.session/org.wso2.carbon.identity.api.user.session.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/session/v1/core/SessionManagementService.java]: done

**Pull request merged**